### PR TITLE
Make modal inputs user-friendly for non-technical users

### PR DESCRIPTION
**Pull request by chiedoclaude**
## Summary
- Replace technical JSON Schema type labels with friendly names visible to users: `string`→`text`, `boolean`→`yes / no`, `array`→`list`, `integer`/`number`→`number`. Types like `object` are hidden entirely.
- Auto-map schema types to appropriate widgets even without explicit `x-ui.widget` hints: `boolean`→toggle switch, `array` (with string items)→list widget, `integer`/`number`→number input.
- Add a new **list widget** for array-type parameters with add/remove item buttons and a clean multi-value UI, replacing the previous plain text input where users had to guess the format.

### Claude Code
- [x] All 943 frontend tests pass
- [x] TypeScript compilation passes with no errors
- [x] Verify no regressions in existing action config and standing approval flows

### Manual Testing
- [ ] Open action config modal for a connector with boolean params (e.g., Stripe `auto_advance`) — should show toggle switch instead of text input
- [ ] Open action config modal for a connector with array params (e.g., Asana `tags`) — should show list widget with Add item button
- [ ] Verify type labels show "text", "yes / no", "number", "list" instead of "string", "boolean", "integer", "array"
- [ ] Verify the standing approval constraint form also uses the new friendly inputs
- [ ] Test adding/removing items in the list widget, then saving the config

https://claude.ai/code/session_018uJE63wn5GKYhZHTe6b1Ry

### CODE REVIEW

File: frontend/src/pages/agents/connectors/ParameterFieldWidget.tsx [hunk 1]
```diff
@@ -202,6 +213,86 @@ function DateTimeWidget({ inputId, value, onChange, disabled, placeholder, class
...
+    }
+  }
+  return [value];
+}
+
+/** Serialize a string array into a JSON array string for form state. */
+function serializeListValue(items: string[]): string {
+  const filtered = items.filter((item) => item !== "");
+  if (filtered.length === 0) return "";
+  return JSON.stringify(filtered);
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/ParameterFieldWidget.tsx]: Good catch — this was a real bug. Fixed in 89537fb: `serializeListValue` no longer filters empty strings, so "Add item" works correctly on empty lists. Empty-string filtering now happens at submission time in `buildParametersFromForm` instead. Also added a test that verifies empty items are filtered at submission.

File: frontend/src/pages/agents/connectors/ParameterFieldWidget.tsx [hunk 2]
```diff
@@ -54,6 +55,14 @@ export function ParameterFieldWidget({
   );
 }
 
+/** Infer widget from schema type when no explicit x-ui.widget is set. */
+function inferWidget(property: SchemaProperty): WidgetType {
+  if (property.type === "boolean") return "toggle";
+  if (property.type === "integer" || property.type === "number") return "number";
+  if (property.type === "array" && property.items?.type === "string") return "list";
+  return "text";
+}
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/ParameterFieldWidget.tsx]: Agreed. Fixed in 89537fb: extracted `inferWidgetFromProperty` as a single exported function in `parameterSchema.ts`. Both the parser (`autoMapWidget`) and the widget component now use the same function — one source of truth.

File: frontend/src/pages/agents/connectors/ActionConfigParameterFields.tsx [hunk 3]
```diff
@@ -177,9 +178,9 @@ function ParameterField({
             required
           </Badge>
         )}
-        {property.type && (
+        {friendlyTypeLabel(property.type) && (
           <span className="text-muted-foreground text-xs">
-            ({property.type})
+            ({friendlyTypeLabel(property.type)})
           </span>
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/ActionConfigParameterFields.tsx]: Fixed in 89537fb: stored the result in a `const typeLabel` in both `ActionConfigParameterFields.tsx` and `SchemaParameterDetails.tsx`.

File: frontend/src/pages/agents/connectors/ParameterFieldWidget.tsx [hunk 4]
```diff
@@ -202,6 +213,86 @@ function DateTimeWidget({ inputId, value, onChange, disabled, placeholder, class
...
+  }
+
+  function addItem() {
+    onChange(serializeListValue([...items, ""]));
+  }
+
+  return (
+    <div className="w-full space-y-2" data-testid={`list-${inputId}`}>
+      {items.map((item, index) => (
+        <div key={index} className="flex items-center gap-2">
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/ParameterFieldWidget.tsx]: Fixed in 89537fb: ListWidget now uses a `useRef`-based counter to generate stable IDs for each row. IDs are assigned when items are added and travel with the item through removal, so React correctly tracks each row's identity.

File: frontend/src/pages/agents/connectors/ParameterFieldWidget.tsx [hunk 5]
```diff
@@ -202,6 +213,86 @@ function DateTimeWidget({ inputId, value, onChange, disabled, placeholder, class
...
+        size="sm"
+        className="gap-1"
+        onClick={addItem}
+        disabled={disabled}
+      >
+        <Plus className="size-4" />
+        Add item
+      </Button>
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/ParameterFieldWidget.tsx:298]: Fixed in 52c10e3: For list widgets, the "Any value" checkbox now sits in the label row (next to the field name and type badge). When checked, the list widget is hidden entirely since there's nothing to configure. This keeps the layout clean for multi-row list inputs.

File: frontend/src/pages/agents/connectors/__tests__/ActionConfigFormFields.test.ts [hunk 6]
```diff
@@ -65,4 +65,16 @@ describe("buildParametersFromForm", () => {
...
+    const result = buildParametersFromForm({ tags: '["a","b"]' }, schema);
+    expect(result).toEqual({ tags: ["a", "b"] });
+  });
+
+  it("falls through to string when array JSON is invalid", () => {
+    const schema = { tags: { type: "array" } };
+    const result = buildParametersFromForm({ tags: "not-json" }, schema);
+    expect(result).toEqual({ tags: "not-json" });
+  });
 });
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/__tests__/ActionConfigFormFields.test.ts:98]: Good catch — this was a real bug, not just a missing test. The `*` pattern auto-detection ran before array type parsing, so `["tag*","billing"]` would get wrapped as a glob pattern instead of being parsed as an array. Fixed in 52c10e3 by moving array parsing before pattern detection. Added the suggested test.

File: frontend/src/pages/agents/connectors/ActionConfigFormFields.tsx [hunk 7]
```diff
@@ -188,6 +188,16 @@ export function buildParametersFromForm(
...
+    } else if (type === "array") {
+      if (value.startsWith("[")) {
+        try {
+          const parsed: unknown = JSON.parse(value);
+          if (Array.isArray(parsed)) {
+            parameters[key] = (parsed as unknown[]).filter((item) => item !== "");
+            continue;
+          }
+        } catch { /* fall through to string */ }
+      }
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/ActionConfigFormFields.tsx]: Fixed in 7504b37: Empty arrays (all blank items) are now omitted from submission instead of being sent as `[]`. The `continue` after filtering ensures the key is skipped entirely when all items are empty, consistent with other widget types. Added test for this case.

File: frontend/src/lib/parameterSchema.ts [hunk 8]
```diff
@@ -89,22 +90,23 @@ export function parseParametersSchema(
...
       const parsedUI = parsePropertyUI(prop["x-ui"]);
-      // Auto-map format: "date-time" → widget: "datetime" when no explicit widget is set
-      const ui =
-        format === "date-time" && !parsedUI?.widget
-          ? { ...parsedUI, widget: "datetime" as const }
-          : parsedUI;
+      // Auto-map types to widgets when no explicit widget is set
+      const ui = autoMapWidget(propType, format, prop, parsedUI);
+      // Preserve items schema for array types
+      const items = parseItems(prop.items);
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/lib/parameterSchema.ts]: Fixed in 7504b37: `parseItems` is now called once per property, and the result is passed into `autoMapWidget` directly instead of being re-parsed inside it.
